src/k8s/pkg/k8sd/database/capi_auth.go

```diff
@@ -0,0 +1,48 @@
+package database
+
+import (
+	"context"
+	"crypto/subtle"
+	"database/sql"
+	_ "embed"
+	"fmt"
+
+	"github.com/canonical/microcluster/cluster"
+)
+
+var (
+	capiTokenStmts = map[string]int{
+		"insert-token": MustPrepareStatement("capi-auth", "insert-token.sql"),
+		"select-token": MustPrepareStatement("capi-auth", "select.sql"),
+	}
+)
+
+// CheckAuthToken returns true if the specified token matches the ClusterAPI token.
+func CheckAuthToken(ctx context.Context, tx *sql.Tx, token string) (bool, error) {
```

Comment:
These will be used as `database.CheckAuthToken` and `database.SetAuthToken`. These names are very generic and do not communicate their purpose well-enough, without the caller having to lookup the implementation details to figure what they are used for. Further, looking for `database.CheckAuthToken()` can be very easily confused with different tokens (e.g. kubernetes tokens, join tokens, etc), which this thing is not.

Instead, consider names like `database.SetClusterAPIToken()` and `database.ValidateClusterAPIToken`, which clearly convey the purpose without callers having to double-check the implementation.